engine/component.ftl

```diff
@@ -126,18 +126,59 @@
 [/#macro]
 
 [#-- Not for general use - framework only --]
-[#assign coreComponentChildConfiguration = [
-    {
-        "Names" : ["Export"],
-        "Types" : ARRAY_OF_STRING_TYPE,
-        "Default" : []
-    },
-    {
-        "Names" : ["DeploymentUnits"],
-        "Types" : ARRAY_OF_STRING_TYPE,
-        "Default" : []
-    }
-] ]
+[#assign coreComponentDeploymentUnitConfiguration = 
+    [
+        {
+            "Names" : ["DeploymentUnits", "deployment:Unit"],
+            "Description" : "An singleton instance of the component, Instance & Version configuration.",
+            "Types" : ARRAY_OF_STRING_TYPE,
+            "Default" : []
+        }
+    ]
+]
+
+[#assign coreComponentChildConfiguration = 
+    [
+        {
+            "Names" : ["Export"],
+            "Types" : ARRAY_OF_STRING_TYPE,
+            "Default" : []
+        },
+        {
+            "Names" : "deployment:Priority",
+            "Description" : "The Deployment Priority",
+            "Types" : NUMBER_TYPE,
+            "Default" : 5
+        },
+        {
+            "Names" : [ "deployment:Group", "DeploymentGroup" ],
+            "Description" : "The component Deployment Group",
+            "Types" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Instances" ],
+            "Description" : "Instances of a component configuration",
+            "SubObjects" : true,
+            "Children" : [
+                {
+                    "Names" : "Versions",
+                    "Description" : "Versions of the components instance.",
+                    "SubObjects" : true,
+                    "Children" : [
+                        {
+                            "Names" : [ "Extensions", "Fragment", "Container" ],
+                            "Description" : "Extensions to invoke as part of component processing",
+                            "Types" : ARRAY_OF_STRING_TYPE,
+                            "Default" : []
+                        }
```

Comment:
Extensions are only supported on select components, not sure why they would need to be explicitly included on Versions. That might be a bug from a test CMDB

Reply:
Ah yep I added this because of an instance of it in one of the provider masterdata files. I'll remove it here and amend the original.